### DISCUSSION TITLE
Do you use total_yield & total_solar_energy

### DISCUSSION THREAD

**Discussion by wills106** [wills106/homeassistant-solax-modbus#79] [Polls]
Through #78 we have realised that sensor.solax_total_solar_energy is in kWh and sensor.solax_total_yield is in MWh and that the scaling for sensor.solax_total_yield is wrong on the Gen4 it should match the Gen3

Do you use sensor.solax_total_yield & sensor.solax_total_solar_energy in any automations or statistics? If so how do you want them to be?

(Scaling will be fixed for Gen4 regardless if they are in kWh or MWh)

**Comment by infradom** [comment]
Please note that we  might also consider 5 additional sensors that cover a  long time: we should also consider 

- battery_input energy total, 
- battery output energy total, 
- grid export total and 
- grid import total, 
- e-charge total

they are currently expressed in kWh

![image](https://user-images.githubusercontent.com/11804014/176898995-511bd3a3-ad88-4e0b-b7c0-c5198e741139.png)


This makes it hard to choose.

**Reply by Kars-de-Jong** [reply]
SolaX Cloud uses kWh.

Personally I don't really care, as long as it plays nice with the energy dashboard.

But having enough precision is important, so if MWh use enough digits.

**Comment by wills106** [comment]
Does anyone else have any views / feedback on what formatting to use. kWh or MWh?

**Comment by Kars-de-Jong** [comment]
Hm, apparently on MIC all measurements are in kWh?

**Reply by wills106** [reply]
They can be set to anything we want in the Integration.

I scaled the Gen2 & Gen3 kWh into MWh to match the SolaX Cloud for the Gen2 & 3.
Although at the Inverter level they are kWh.

I am tempted to have them all kWh.
But people don't read release notes and I end up with 3-4 issues for the same issue, that isn't even and issue if they read the release notes / the Readme!

**Comment by wills106** [comment]
Changed to be kWh in 0.6.0 Beta 8
